src/ray/raylet/worker_pool.h

```diff
@@ -40,13 +41,30 @@ namespace raylet {
 using WorkerCommandMap =
     std::unordered_map<Language, std::vector<std::string>, std::hash<int>>;
 
-enum class RuntimeEnvStatus {
-  /// This runtime env is currently being installed.
-  PENDING,
-  /// This runtime env has completed installation (either successfully or not)
-  DONE
+enum PopWorkerStatus {
+  // OK.
+  // A registered worker will be returned with callback.
+  OK = 0,
+  // Job config is not found.
+  // A nullptr worker will be returned with callback.
+  JobConfigMissing = 1,
+  // Worker process startup rate is limited.
+  // A nullptr worker will be returned with callback.
+  TooManyStartingWorkerProcesses = 2,
+  // Worker process has been started, but the worker did not register at the raylet within
+  // the timeout.
+  // A nullptr worker will be returned with callback.
+  WorkerPendingRegistration = 3,
+  // Any fails of runtime env creation.
+  // A nullptr worker will be returned with callback.
+  RuntimeEnvCreationFailed = 4,
 };
 
+/// \Return True if any task dispatched successfully to the worker. Otherwise, return
+/// false.
```

Comment:
`/// \Return true if the worker was used. Otherwise, return false and the worker will be returned to the worker pool.`
Because from the worker pool's perspective, it should only care the worker, not the task.